src/app/loadout-builder/process/useProcess.ts

```diff
@@ -251,18 +264,22 @@ function groupItems(
       defs && upgradeSpendTierToMaxEnergy(defs, upgradeSpendTier, item) === 10
     }`;
 
-    if (raidCombatAndLegacyMods.length) {
+    if (activityMods.length) {
       groupId += `${getSpecialtySocketMetadatas(item)
         ?.map((metadata) => metadata.slotTag)
         .join(',')}`;
     }
 
     // We don't need to worry about slot specific energy as items are already filtered for that.
     if (
-      someModHasEnergyRequirement(raidCombatAndLegacyMods) ||
-      someModHasEnergyRequirement(generalMods)
+      someModHasEnergyRequirement(combatMods) ||
+      someModHasEnergyRequirement(generalMods) ||
+      someModHasEnergyRequirement(activityMods)
     ) {
-      groupId += `${item.energy?.energyType}`;
+      groupId +=
+        defs && canSwapEnergyFromUpgradeSpendTier(defs, upgradeSpendTier, item, lockItemEnergyType)
+          ? DestinyEnergyType.Any
```

Comment:
So is this the only actual change to the logic, then? That we'll ignore energy requirements if we're allowed to swap energy?

Reply:
Yes that is it, before we were always creating groups in class items of each energy type if you locked mods with an energy requirement. Really we should only care about energy types that we cannot change.